packages/client/src/forms/configuration/index.ts

```diff
@@ -384,11 +386,20 @@ export function configureRegistrationForm(
         /*
          * Splice the disabled field away only when in registration, not in configuration mode
          */
-        newForm.sections[
-          defaultPropCustomisation.defaultField.selectedSectionIndex
-        ].groups[
-          defaultPropCustomisation.defaultField.selectedGroupIndex
-        ].fields.splice(defaultPropCustomisation.defaultField.index, 1)
+
+        // update preceedingDefaultField.index as it may have changed now that custom groups are added
+        let newIndex = 0
+        groupFields.filter((field, index) => {
```

Comment:
```suggestion
        groupFields.forEach((field, index) => {
```
`findIndexOf` would probably be the better choice here.